packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -1165,27 +1152,42 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
   Widget build(BuildContext context) {
     updateScrollbarPainter();
 
-    Widget child = CustomPaint(
-      key: _scrollbarPainterKey,
-      foregroundPainter: scrollbarPainter,
-      child: RepaintBoundary(child: widget.child),
-    );
-
-    if (!_isMobile) {
-      // Hover events not supported on mobile.
-      child = MouseRegion(
-        onExit: handleHoverExit,
-        onHover: handleHover,
-        child: child
-      );
-    }
-
     return NotificationListener<ScrollNotification>(
       onNotification: _handleScrollNotification,
       child: RepaintBoundary(
         child: RawGestureDetector(
           gestures: _gestures,
-          child: child,
+          child: MouseRegion(
+            onExit: (PointerExitEvent event) {
+              switch(event.kind) {
+                case PointerDeviceKind.mouse:
+                  handleHoverExit(event);
+                  break;
+                case PointerDeviceKind.stylus:
+                case PointerDeviceKind.invertedStylus:
+                case PointerDeviceKind.unknown:
+                case PointerDeviceKind.touch:
+                  break;
+              }
```

Comment:
Why not,
```dart
if (event.kind == PointerDeviceKind.mouse)
  handleHoverExit(event);
```

Reply:
Cleaner for sure! I think we usually use switches for enums, it makes sure we don't miss a case in future changes. https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values